exercises/robot-simulator/src/test/java/RobotTest.java

```diff
@@ -0,0 +1,211 @@
+        final Robot robot = new Robot(new GridPosition(0, 0), Orientation.SOUTH);
+
+        robot.advance();
+
+        final GridPosition expectedGridPosition = new GridPosition(0, -1);
+        final GridPosition actualGridPosition = robot.getGridPosition();
+
+        assertTrue(actualGridPosition.equals(expectedGridPosition));
+    }
+
+    @Ignore
+    @Test
+    public void testAdvancingWhenFacingEastIncreasesXCoordinateByOne() {
+        final Robot robot = new Robot(new GridPosition(0, 0), Orientation.EAST);
+
+        robot.advance();
+
+        final GridPosition expectedGridPosition = new GridPosition(1, 0);
+        final GridPosition actualGridPosition = robot.getGridPosition();
+
+        assertTrue(actualGridPosition.equals(expectedGridPosition));
+    }
+
+    @Ignore
+    @Test
+    public void testAdvancingWhenFacingWestDecreasesXCoordinateByOne() {
+        final Robot robot = new Robot(new GridPosition(0, 0), Orientation.WEST);
+
+        robot.advance();
+
+        final GridPosition expectedGridPosition = new GridPosition(-1, 0);
+        final GridPosition actualGridPosition = robot.getGridPosition();
+
+        assertTrue(actualGridPosition.equals(expectedGridPosition));
+    }
+
+    @Ignore
+    @Test
+    public void testInstructionsToMoveWestAndNorth() {
+        testInstructions(
+                new GridPosition(0, 0),
+                Orientation.NORTH,
+                "LAAARALA",
+                new GridPosition(-4, 1),
+                Orientation.WEST);
+    }
+
+    @Ignore
+    @Test
+    public void testInstructionsToMoveWestAndSouth() {
+        testInstructions(
+                new GridPosition(2, -7),
+                Orientation.EAST,
+                "RRAAAAALA",
+                new GridPosition(-3, -8),
+                Orientation.SOUTH);
+    }
+
+    @Ignore
+    @Test
+    public void testInstructionsToMoveEastAndNorth() {
+        testInstructions(
+                new GridPosition(8, 4),
+                Orientation.SOUTH,
+                "LAAARRRALLLL",
+                new GridPosition(11, 5),
+                Orientation.NORTH);
+    }
+
+    // Helper Methods
+
+    private void testTurningRight(final Orientation startOrientation, final Orientation expectedEndOrientation) {
+        final Robot robot = new Robot(new GridPosition(0, 0), startOrientation);
+
+        robot.turnRight();
+
+        final Orientation actualEndRobotOrientation = robot.getOrientation();
+
+        assertTrue(actualEndRobotOrientation.equals(expectedEndOrientation));
+    }
+
+    private void testTurningLeft(final Orientation startOrientation, final Orientation expectedEndOrientation) {
+        final Robot robot = new Robot(new GridPosition(0, 0), startOrientation);
+
+        robot.turnLeft();
+
+        final Orientation actualEndRobotOrientation = robot.getOrientation();
+
+        assertTrue(actualEndRobotOrientation.equals(expectedEndOrientation));
+    }
+
+    private void testInstructions(
```

Comment:
Personally, I'd inline them.

To come to that conclusion I did this little exercise with myself: I pretended I was seeing this exercise for the first time and just myopically walked through the test methods from the top.  When I got to `testTurningRightCorrectlyChangesOrientation()`, I had to then jump down to the definition of `testTurningRight()` which, on examination, didn't include any more "boilerplate" than other test methods... at which point the whiny pedantic voice in my head said, "why not just include that in the test above?"